src/argilla/client/client.py

```diff
@@ -104,11 +104,28 @@ def __init__(
                 the headers of argilla client requests, like additional security restrictions. Default: `None`.
 
         """
-        api_url = api_url or os.getenv("ARGILLA_API_URL", "http://localhost:6900")
+        from argilla.client.login import ArgillaCredentials
+
+        api_url = api_url or os.getenv("ARGILLA_API_URL")
+        api_key = api_key or os.getenv("ARGILLA_API_KEY")
+        workspace = workspace or os.getenv("ARGILLA_WORKSPACE")
+        extra_headers = extra_headers or {}
+
+        if api_url is None and api_key is None and workspace is None:
```

Comment:
Since the `workspace` is optional and has nothing to do with the connection to Argilla itself, shouldn't we just remove the condition from here? e.g. we store the credentials for the `owner` and `workspace=argilla`, but we want to connect to it using `workspace=gabriel`, then we won't be able to reuse those values, right? But we can indeed just assign a different workspace via the same `api_url` and `api_key`, does that make sense?